From 9dd20cabea2b22a44948ce9dac4dd3213fa5ccfc Mon Sep 17 00:00:00 2001
From: Su Yue <glass@fydeos.io>
Date: Sat, 17 Sep 2022 20:37:08 +0800
Subject: [PATCH 22/38] Switch to slot b after all things done

Signed-off-by: Su Yue <glass@fydeos.io>
---
 py/gooftool/commands.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/py/gooftool/commands.py
+++ b/py/gooftool/commands.py
@@ -53,6 +53,7 @@
 from cros.factory.utils.debug_utils import SetupLogging
 from cros.factory.utils import file_utils
 from cros.factory.utils.process_utils import Spawn
+from cros.factory.utils.process_utils import CheckOutput
 from cros.factory.utils import sys_interface
 from cros.factory.utils import sys_utils
 from cros.factory.utils import time_utils
@@ -1285,6 +1286,8 @@
   wipe_args = PrepareWipeArgs(options)
 
   # ExecFactoryPar('gooftool', WIPE_IN_PLACE, *wipe_args)
+  p = CheckOutput("/usr/local/factory/bin/fydeos_factory.sh mark_partition_bootable b", shell=True, log=True)
+  logging.info(p)
 
 
 @Command('get_firmware_hash',
