From 74f81087e134573035fec69ffa355ce492a9d77e Mon Sep 17 00:00:00 2001
From: Su Yue <glass@fydeos.io>
Date: Mon, 19 Sep 2022 09:41:57 +0800
Subject: [PATCH 30/38] Hardcode fan speed to 0

Signed-off-by: Su Yue <glass@fydeos.io>
---
 py/test/pytests/countdown.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/py/test/pytests/countdown.py
+++ b/py/test/pytests/countdown.py
@@ -317,7 +317,7 @@
 
   def SnapshotStatus(self):
     return Status(self._dut.thermal.GetAllTemperatures(),
-                  self._dut.fan.GetFanRPM(),
+                  0,
                   self._cpu_freq_manager.GetCurrentFrequency())
 
   def setUp(self):
