From 03820af63709de3a296fca01553210b5a378ecb2 Mon Sep 17 00:00:00 2001
From: Su Yue <glass@fydeos.io>
Date: Mon, 19 Sep 2022 12:28:44 +0800
Subject: [PATCH 38/38] Call reboot at the end

Signed-off-by: Su Yue <glass@fydeos.io>
---
 py/gooftool/commands.py | 1 +
 1 file changed, 1 insertion(+)

--- a/py/gooftool/commands.py
+++ b/py/gooftool/commands.py
@@ -1288,6 +1288,7 @@
   # ExecFactoryPar('gooftool', WIPE_IN_PLACE, *wipe_args)
   p = CheckOutput("/usr/local/factory/bin/fydeos_factory.sh mark_partition_bootable b", shell=True, log=True)
   logging.info(p)
+  p = CheckOutput("/sbin/reboot", shell=True, log=False)
 
 
 @Command('get_firmware_hash',
